packages/@aws-cdk/aws-logs-destinations/lib/lambda.ts

```diff
@@ -24,13 +24,15 @@ export class LambdaDestination implements logs.ILogSubscriptionDestination {
   public bind(scope: Construct, logGroup: logs.ILogGroup): logs.LogSubscriptionDestinationConfig {
     const arn = logGroup.logGroupArn;
     if (this.options.addPermissions !== false) {
-      this.fn.addPermission('CanInvokeLambda', {
+      const permissionId = 'CanInvokeLambda';
+      this.fn.addPermission(permissionId, {
         principal: new iam.ServicePrincipal('logs.amazonaws.com'),
         sourceArn: arn,
         // Using SubScription Filter as scope is okay, since every Subscription Filter has only
         // one destination.
         scope,
       });
+      scope.node.addDependency(scope.node.findChild(permissionId));
```

Comment:
I think we should use `tryFindChild` instead. I don't think we want to error out if the permission isn't created.